kythe/go/serving/xrefs/xrefs.go

```diff
@@ -989,11 +1120,11 @@ func a2a(a *srvpb.ExpandedAnchor, fileInfos map[string]*srvpb.FileInfo, anchorTe
 		Snippet:     a.Snippet,
 		SnippetSpan: a.SnippetSpan,
 		BuildConfig: a.BuildConfiguration,
-		Revision:    revision,
+		Revision:    fileInfo.GetRevision(),
 	}}
 }
 
-func d2d(d *srvpb.Document, patterns []*regexp.Regexp, nodes map[string]*cpb.NodeInfo, defs map[string]*xpb.Anchor, fileInfos map[string]*srvpb.FileInfo) *xpb.DocumentationReply_Document {
+func d2d(d *srvpb.Document, patterns []*regexp.Regexp, nodes map[string]*cpb.NodeInfo, defs map[string]*xpb.Anchor, fileInfos map[string]*srvpb.FileInfo, p patcherFunc) *xpb.DocumentationReply_Document {
```

Comment:
idly wondering if a2a / d2d might be clearer if they had a receiver type that encapsulated the regexp, nodeinfo, anchors, etc.